src/utils/dom.js

```diff
@@ -78,3 +79,18 @@ export function focus(root: ?HTMLElement, selector?: string, focusRoot: boolean
         root.focus();
     }
 }
+
+export function scrollItemIntoView(id: ?string, className: string): void {
```

Comment:
Probably can take in array of classNames if you want to test more than one.
You can also use element.closest instead of the loop if its available. You can check the prototype if its available https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#Polyfill

Reply:
Also, should the function just take an HTML element instead of ID?
```
function scrollIntoView(element: ?HTMLElement, classNames: Array<string>)
```

Reply:
probably also see if you can add a small unit test for this

Reply:
@priyajeet, is it a good idea to polyfill .closest? If not, let's consider adding to utils/dom/closest. @ChenCodes, the source code is provided in the [MDN link](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest#Polyfill).

RE: HTML Element instead of ID. I vote no. This function is looking for the closest element of the target classnames (It doesn't yet but should implement that. I've given feedback here). It behaves more like .closest. See the previous implementation here:

```            while (
                parentEl.parentElement instanceof HTMLElement &&
                !parentEl.classList.contains('scroll-wrap-container') && // Scroll the container, not the body when contained in a scroll-container.
                !parentEl.classList.contains('modal') // Scroll the modal, not the body when contained in a modal.
            ) {
                parentEl = parentEl.parentElement;
            }
```

I propose doing something like this instead of the while-loop:
```const parentEl = el.closest('html, .modal, .overlay-wrapper');
```